core-plugins/src/test/java/co/cask/hydrator/plugin/batch/joiner/JoinerConfigTest.java

```diff
@@ -184,4 +184,25 @@ public void testJoinerConfigWithDuplicateOutputFields() {
     Joiner joiner = new Joiner(config);
     joiner.configurePipeline(pipelineConfigurer);
   }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testJoinerConfigWithInvalidJoinKeySchema() {
+    String selectedFields = "film.film_id, film.film_name, " +
+      "filmActor.actor_name as renamed_actor, filmCategory.category_name as renamed_category";
+
+    Schema filmCategorySchema = Schema.recordOf(
+      "filmCategory",
+      Schema.Field.of("film_id", Schema.of(Schema.Type.LONG)),
+      Schema.Field.of("film_name", Schema.of(Schema.Type.STRING)),
+      Schema.Field.of("category_name", Schema.of(Schema.Type.STRING)));
+
+    JoinerConfig config = new JoinerConfig("film.film_id=filmActor.film_id=filmCategory.film_id&" +
+                                             "film.film_name=filmActor.film_name=filmCategory.film_name",
+                                           selectedFields, "film,filmActor,filmCategory");
+    MockPipelineConfigurer pipelineConfigurer = new MockPipelineConfigurer(
```

Comment:
we should avoid adding new usages of this class and eventually remove it, as it guarantees that whenever the Hydrator PipelineConfigurer API changes, this repo will break.

I think we can just make validateJoinKeySchemas package private and call it directly from this class.